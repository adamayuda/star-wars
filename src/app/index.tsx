import "./style.sass";

import React, { FC, Suspense, lazy } from "react";

import { Route, RouteProps, Switch } from "react-router-dom";

import { Loading } from "src/components/loading";

/* eslint-disable  @typescript-eslint/no-explicit-any */
interface RouteInterface extends RouteProps {
  import: Promise<any>;
}

const routes: RouteInterface[] = [
  {
    import: import("src/pages/home"),
    path: "/",
    exact: true,
  },
  {
    import: import("src/pages/login"),
    path: "/login",
    exact: true,
  },
];

export const App: FC = () => {
  return (
    <Suspense fallback={<Loading />}>
      <Switch>
        {routes.map((route, index: number) => (
          <Route
            {...route}
            key={`route-${index}`}
            component={lazy(() => route.import)}
          />
        ))}
      </Switch>
    </Suspense>
  );
};
