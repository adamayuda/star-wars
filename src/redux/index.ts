import { applyMiddleware, compose, createStore } from "redux";
import { mainReducer } from "src/redux/reducers";
import thunk from "redux-thunk";

const composeEnhancers =
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  (window as any).__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose;

export const mainStore = createStore(
  mainReducer,
  composeEnhancers(applyMiddleware(thunk)),
);

export const createMainStore = () => {
  return createStore(mainReducer, composeEnhancers(applyMiddleware(thunk)));
};

export type StateInterface = ReturnType<typeof mainStore.getState>;

export type ActionType = "USER_LOGIN";

export interface Action<T> {
  type: ActionType;
  payload: Partial<T>;
}
