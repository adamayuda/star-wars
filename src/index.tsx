import { App } from "src/app";
import { Provider } from "react-redux";

import React from "react";
import ReactDOM from "react-dom";

import { BrowserRouter as Router } from "react-router-dom";

import { mainStore } from "src/redux";

ReactDOM.render(
  <Provider store={mainStore}>
    <Router>
      <App />
    </Router>
  </Provider>,
  document.getElementById("root"),
);
