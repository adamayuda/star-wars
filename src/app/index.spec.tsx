import { cleanup, render } from "@testing-library/react";

import { App } from ".";
import { Provider } from "react-redux";
import React from "react";

import { BrowserRouter as Router } from "react-router-dom";
import { createMainStore } from "src/redux";

let mainStore = createMainStore();
global.fetch = jest.fn();

describe("src/app/index.tsx", () => {
  afterEach(() => cleanup());
  beforeEach(() => {
    global.fetch.mockImplementation(
      () =>
        new Promise((resolve) => {
          resolve({
            json: () =>
              new Promise((resolve) =>
                resolve({
                  picture: "Post litle",
                  username: "Post Boby",
                }),
              ),
          });
        }),
    );
    mainStore = createMainStore();
  });

  test("Render App page", async () => {
    const renderedApp = render(
      <Provider store={mainStore}>
        <Router>
          <App />
        </Router>
      </Provider>,
    );

    expect(renderedApp.container).toMatchSnapshot();
  });

  test("Render App page with laptop device", async () => {
    mainStore.dispatch({
      type: "UPDATE_DEVICE_TYPE",
      payload: { type: "laptop" },
    });

    const renderedApp = render(
      <Provider store={mainStore}>
        <Router>
          <App />
        </Router>
      </Provider>,
    );

    expect(renderedApp.container).toMatchSnapshot();
  });
});
