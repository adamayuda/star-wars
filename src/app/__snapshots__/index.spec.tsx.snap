// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/app/index.tsx Render App page 1`] = `
<div>
  <div
    class="loading"
  >
    <img
      alt="LinkedTeam"
      src="test-file-stub"
    />
  </div>
</div>
`;

exports[`src/app/index.tsx Render App page with laptop device 1`] = `
<div>
  <div
    class="loading"
  >
    <img
      alt="LinkedTeam"
      src="test-file-stub"
    />
  </div>
</div>
`;
